compiler/rustc_mir_dataflow/src/framework/engine.rs

```diff
@@ -181,37 +181,61 @@ where
         self
     }
 
-    /// Computes the fixpoint for this dataflow problem and returns it.
+    /// Computes the fixpoint for this dataflow problem over all basic blocks and returns the
+    /// fixpoint.
     pub fn iterate_to_fixpoint(self) -> Results<'tcx, A>
     where
         A::Domain: DebugWithContext<A>,
     {
-        let Engine {
-            analysis,
-            body,
-            dead_unwinds,
-            mut entry_sets,
-            tcx,
-            apply_trans_for_block,
-            pass_name,
-            ..
-        } = self;
+        let mut dirty_queue: WorkQueue<BasicBlock> =
+            WorkQueue::with_none(self.body.basic_blocks().len());
+
+        self.body.basic_blocks().indices().for_each(|bb| {
+            dirty_queue.insert(bb);
+        });
```

Comment:
The order in which blocks are added to the work queue is significant. I suspect this will hurt performance. It would be kind of fun to check though, since the basic block numbering is usually fairly close to RPO due to how we build MIR.